### PR TITLE
Kumarde/zlint refactor

### PR DESCRIPTION
**Pull request by kumarde**
Based on a conversation w/ @zakird , we want to keep tracking of lints in a struct rather than a dynamic map so we catch errors at compile time rather than runtime. 

Looking for feedback on the best way to do this. I don't want to couple the lint code to a specific struct field, so right now I'm just checking the name and updating the appropriate field, but not sure if there's a better to handle this.

### CODE REVIEW

**Comment by zakird** [comment]
This really doesn't seem like it solves any problems, it just moves them
around. But I understand how we arrived here too based on how lints are
registered. Otherwise, we have to do this in a much more static method. But
given that the protobuf is static, it's not like we're gaining anything by
having run time registration, as evidenced by that function....

I'm not 100% sure how one is supposed to do this in golang. @dadrian do you
have thoughts?

**Comment by zakird** [comment]
It's also worth noting that his solution is probably slower. It's O(n^2)
whereas the old solution was O(n) with out any downfalls compared to this
solution.

**Comment by kumarde** [comment]
@zakird yeah you're right. I could use a static map I guess to do this, but I wonder if that's _really_ better than the old solution.

**Comment by dadrian** [comment]
I don't fully have enough context, what type of errors do you want caught at compile-vs-run time? What is the problem we're trying to solve?

**Comment by kumarde** [comment]
@dadrian when ZLint is run over a certificate, it currently returns a `map[string]string`, indexed by lint_name. @zakird mentioned to me that's dangerous because we won't know at compile time whether or not indexing into that map will fail, and so it's better to expose a struct that requires the calling code to directly reference fields rather than indexing into a map. Unfortunately, we register lints at runtime and in order to do that, we just need to do the string compare in the ZLint library rather than the calling application.

So the question: do you have any suggestions on how to do this idiomatically in Golang?

File: util/fqdn_test.go [hunk 1]
```diff
@@ -31,7 +31,7 @@ func TestIsFQDNQuestionMarkFQDN(t *testing.T) {
 }
 
 func TestIsFQDNQuestionMarkIncorrectPlaceFQDN(t *testing.T) {
-	domain := "?.?.abc.?com"
+	domain := "?.?.abc?.com"
```
Comment (zakird) [inline util/fqdn_test.go]: These seem equally incorrect. wat.

Comment (kumarde) [inline util/fqdn_test.go]: I think govalidator stopped checking the suffix for things; I just pulled the update and a lot of changes were made. _sigh_

Comment (zakird) [inline util/fqdn_test.go]: wat?

Comment (kumarde) [inline util/fqdn_test.go]: I have literally no idea why this test was all of a sudden breaking. Something changed in the underlying govalidator library that we rely on to check if something is an FQDN.

**Comment by zakird** [comment]
What's the current status of this?

**Comment by kumarde** [comment]
@zakird ready for code review

**Pull request merged**